subprojects/resources/src/main/java/org/gradle/internal/resource/metadata/ExternalResourceMetaData.java

```diff
@@ -58,4 +58,14 @@
     @Nullable
     HashValue getSha1();
 
+    /**
+     * When the item should be evicted from the cache.
+     *
+     * This should only be collected from HTTP Headers. When the HTTP request does not include it, the value will be null.
+     *
+     * @return Cache expiration from HTTP headers, null if not specified or not from an HTTP request.
+     */
+    @Nullable
+    Date getValidUntil();
```

Comment:
Would be better to return a unix epoch long here instead of a Date. Date is mutable and has ambiguous TZ semantics.